[10.0] migrate base_gs1_barcode

Cedric-Pigeon opened the pull request.

Commit:
[10.0] migrate base_gs1_barcode

Commit:
[ADD] add setup

colmeirin commented:
Hello, does this work as a base for a 9.0 migration?
Regards

alexis-via commented on base_gs1_barcode/__manifest__.py:
```diff
@@ -1,70 +1,49 @@
-    'installable': False,
-    'license': 'GPL-3',
+    'installable': True,
+    'license': 'AGPL-3',
```
Do you have the agreement of Numérigraphe to change the licence from GPL to AGPL ?

alexis-via commented on base_gs1_barcode/models/gs1_barcode.py:
```diff
@@ -1,80 +1,80 @@
-from openerp.osv import orm, fields
-from openerp.tools.translate import _
+from odoo import api, fields, models
+from odoo.tools.translate import _
```
You can do : from odoo import api, fields, models, _
It saves one line :)

alexis-via commented on base_gs1_barcode/views/gs1_barcode_view.xml:
```diff
@@ -41,6 +41,6 @@
             <field name="view_mode">tree,form</field>
         </record>
-        <menuitem action="action_config_gs1_barcode" id="menu_config_gs1_barcode" parent="product.prod_config_main" />
+        <menuitem action="action_config_gs1_barcode" id="menu_config_gs1_barcode"/>
```
what about the parent parameter ?

alexis-via commented:
FYI, the date parsing when the day = 00 is wrong: https://www.gs1.org/docs/barcodes/GS1_General_Specifications.pdf page 143 section 3.4.7 first note:
```
Note : When it is not necessary to specify the day (the day field is filled with two zeros), the 
resultant data string SHALL be interpreted as the last day of the noted month including any 
adjustment for leap years (e.g., “130200” is “2013 February 28”, “160200” is “2016 February 
29”, etc.).
```
It's a small detail, but...

legalsylvain commented on base_gs1_barcode/__manifest__.py:
```diff
@@ -1,70 +1,49 @@
     'category': 'Generic Modules/Inventory Control',
     'description': """
-This module provides an API to decoding the content of structured barcodes \
+This module provides an API to decoding the content of structured barcodes
```
could you move all the description into the readme file ?

gurneyalex replied:
yes please

legalsylvain commented on base_gs1_barcode/test/gs1_barcode_test.yml:
```diff
@@ -1,22 +0,0 @@
-- Test the decoding API for GS1-128 barcodes
```
why did you removed the test ?

gurneyalex reviewed (changes requested):
Please revert licence change (unless you have an agreement from all contributors that they agree).

alexis-via committed:
base_gs1_barcode: Revert to GPL-3 licence

Fix views and add menu entry under Settings > Technical
Comply with GS1 specs for date parsing when day = 00
Better ACL
Code cleanup

alexis-via committed:
Fix date parsing when day = 00

alexis-via commented:
FYI, I made a PR to acsone's branch to update this PR https://github.com/acsone/stock-logistics-barcode/pull/2

Cedric-Pigeon commented:
@alexis-via I merged your PR. thanks a lot

alexis-via commented:
@gurneyalex licence change has been reverted ; could you remove your "change request" ?

rafaelbn commented:
Please @Cedric-Pigeon check travis, is :red_circle:

rousseldenis commented:
@numerigraphe @clonedagain Do you see objection to AGPL change ?

rousseldenis committed:
[10.0][IMP] base_gs1_barcode: Improve README, manifest

clonedagain commented:
@rousseldenis sorry your message had not reached me until today.
I need to ask the boss before I can agree officially.
I'll get back to you soon.

rousseldenis commented:
@clonedagain No problem. This is to have a clear answer on that point before merging.

clonedagain commented:
@rousseldenis sorry for the delay about the re-licensing. The person in charge of IP admits its a subtle change and he needs time to consider it.
Maybe someone from OCA can give us a call (in French if at all possible), if you really need our approval ?

That being said, doesn't the OCA CLA give you power to proceed with the change already ?

rousseldenis commented:
@clonedagain It was just to know if we wait before merge or merge like this
